napari/layers/_vectors_layer/model.py

```diff
@@ -100,8 +109,10 @@ def range(self):
             maxs = np.ones(self.data.ndim, dtype=int)
             mins = np.zeros(self.data.ndim, dtype=int)
         else:
-            maxs = np.max(self.data[:, 0, :], axis=0) + 1
-            mins = np.min(self.data[:, 0, :], axis=0)
+            data = copy(self.data)
+            data[:, 1, :] = data[:, 0, :] + self.length * data[:, 1, :]
+            maxs = np.max(data, axis=(0, 1)) + 1
+            mins = np.min(data, axis=(0, 1)) + 1
```

Comment:
This looks very reminiscent of the above. Perhaps abstract it into a function? I guess this is converting from `src, dst` coordinates to `src, proj` coordinates? A function could document the exact purpose of these lines and make sure that any future tweaks are applied uniformly everywhere.

Reply:
I've removed the code above to reduce duplication and now added a comment. some of this will likely get cleaned up in refactoring all the layers. in many ways the above `_get_shape` has been superseded by `range` and is no longer needed